clouddriver-configserver/src/main/java/com/netflix/spinnaker/clouddriver/config/ConfigurationProvider.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.config;
+
+import java.util.Map;
+import org.springframework.boot.context.properties.bind.BindResult;
+import org.springframework.core.env.PropertySource;
+
+/**
+ * This interface is meant to provide a custom way of handling configuration properties. Usually,
+ * configuration properties are handled implicitly by Spring boot itself. But in case that is not
+ * desired, this will allow it to be handled in a custom way where properties are read from a
+ * property source, then converted to a map as an intermediate step which is then bound to a target
+ * implementation class.
+ *
+ * <p>This interface defines the necessary actions which would be required to do the same.
+ */
+public interface ConfigurationProvider {
+  /**
+   * Returns the desired configuration properties, bound to the target implementation class.
+   *
+   * @return a target implementation class for a property
+   */
+  Object getConfigurationProperties();
```

Comment:
I think it will work better if you use generics instead of `Object`, so the interface definition would be something like
```
public interface ConfigurationProvider<T> {
   T getConfigurationProperties();
   ...
}
```

Reply:
fixed